docs/software_requirements/system_initialization.sdoc

```diff
@@ -10,3 +10,269 @@ IMPORT_FROM_FILE: software_requirements.sgra
 TBD
 
 [/FREETEXT]
+
+[SECTION]
+TITLE: Interface Requirements
+
+[FREETEXT]
+
+TBD
+
+[/FREETEXT]
+
+[REQUIREMENT]
+UID: ZEP-127
+STATUS: Draft
+TYPE: Interface
+COMPONENT: System Initialization
+TITLE: Build-time Registration for System Initialization Notifications
+STATEMENT: >>>
+The Zephyr RTOS shall expose an interface to build-time registration for the following system initialization notifications:
+
+- Pre-CPU Architecture Initialization
+- Pre-Kernel Initialization
+- Post-Kernel Initialized
+- Application Initialization
+<<<
+
+[REQUIREMENT]
+UID: ZEP-143
+STATUS: Draft
+TYPE: Interface
+COMPONENT: System Initialization
+TITLE: Build-time Registration for SMP-Initialized Notification
+STATEMENT: >>>
+Where the Symmetric Multi-Processing feature is enabled,
+
+the Zephyr RTOS shall expose an interface to build-time registration for the following system initialization notifications:
+
+- Symmetric Multi-Processing Initialized.
+<<<
+
+[/SECTION]
+
+[SECTION]
+TITLE: Functional Requirements
+
+[FREETEXT]
+
+TBD
+
+[/FREETEXT]
+
+[REQUIREMENT]
+UID: ZEP-128
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Sustain hardware watchdog inherited from preceding executable
+STATEMENT: >>>
+Where the Zephyr RTOS inherits an active hardware watchdog that cannot be reconfigured,
+
+the Zephyr RTOS shall sustain the active hardware watchdog.
+<<<
+
+[REQUIREMENT]
+UID: ZEP-129
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Initial CPU stack usage kept within published limits
+STATEMENT: >>>
+The Zephyr RTOS shall use no more than the published amount of the initial CPU stack.
+<<<
+
+[REQUIREMENT]
+UID: ZEP-130
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Preceding executable data availability
+STATEMENT: >>>
+While the Zephyr RTOS is in any system initialization phase,
+
+the Zephyr RTOS shall allow read access to the data provided by the preceding executable.
+<<<
+
+[REQUIREMENT]
+UID: ZEP-131
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Initialization follows component dependency order
+STATEMENT: >>>
+The Zephyr RTOS shall initialize the system following the order of component dependencies.
+<<<
+
+[REQUIREMENT]
+UID: ZEP-138
+STATUS: Draft
+TYPE: Functional
+COMPONENT: System Initialization
+TITLE: Initialization ends with call to `main(void)`
+STATEMENT: >>>
+When all of the following has been established,
+
+the Zephyr RTOS shall call the C function `main(void)` on the system initialization stack:
+
+- each C++ static object has been initialized; and
+- each SMP context has been initialized; and
+- each System Initialization notification has been published; and
+- each memory access protection has been established.
+<<<
+
```

Comment:
Which sdoc fragment should hold the requirements for the default implementation of `int main(void);`?